subsys/bluetooth/controller/hci/hci.c

```diff
@@ -6246,31 +6314,39 @@ static void le_per_adv_sync_report(struct pdu_data *pdu_data,
 		BT_DBG("    AD Data (%u): <todo>", data_len);
 	}
 
+	if (0) {
+
 #if (CONFIG_BT_CTLR_DUP_FILTER_LEN > 0) && \
 	defined(CONFIG_BT_CTLR_SYNC_PERIODIC_ADI_SUPPORT)
-	if (IS_ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC_ADI_SUPPORT) && adi) {
+	} else if (IS_ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC_ADI_SUPPORT) &&
+		   adi) {
 		const struct ll_sync_set *sync = HDR_LLL2ULL(ftr->param);
-
-		/* FIXME: Use correct data status else chain PDU report will
-		 *        be filtered out.
-		 */
-		dup = sync->nodups && dup_found(PDU_ADV_TYPE_EXT_IND,
-						sync->peer_id_addr_type,
-						sync->peer_id_addr,
-						DUP_EXT_ADV_MODE_PERIODIC,
-						adi, 0U);
-	}
+		uint8_t data_status;
+
+		data_status = (aux_ptr) ?
+			      BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_PARTIAL :
+			      BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_COMPLETE;
+
+		drop = !ftr->sync_rx_enabled ||
+		       (sync->nodups && dup_found(PDU_ADV_TYPE_EXT_IND,
+						  sync->peer_id_addr_type,
+						  sync->peer_id_addr,
+						  DUP_EXT_ADV_MODE_PERIODIC,
+						  adi, data_status));
```

Comment:
Consider moving the result of `dup_found` to a separate variable for readability.